client/extensions/woocommerce/app/products/product-create.js

```diff
@@ -39,19 +39,21 @@ class ProductCreate extends React.Component {
 	componentDidMount() {
 		const { product, siteId } = this.props;
 
-		if ( ! product ) {
-			this.props.editProduct( null, {
-				type: 'simple'
-			} );
-		}
-
 		if ( siteId ) {
```

Comment:
Could you add a TODO comment for us to deliberately handle a situation where we don't have a `siteId` for this page? I feel this might be possible to occur at least with a hand-written URL.